spring-integration-amqp/src/main/java/org/springframework/integration/amqp/inbound/AmqpInboundGateway.java

```diff
@@ -67,14 +87,17 @@ public AmqpInboundGateway(AbstractMessageListenerContainer listenerContainer) {
 						"the adapter needs to configure its own listener implementation.");
 		this.messageListenerContainer = listenerContainer;
 		this.messageListenerContainer.setAutoStartup(false);
-		this.amqpTemplate = new RabbitTemplate(this.messageListenerContainer.getConnectionFactory());
+		this.amqpTemplate = amqpTemplate;
+		this.amqpTemplateExplicitlySet = amqpTemplateExplicitlySet;
 	}
 
 
 	public void setMessageConverter(MessageConverter messageConverter) {
 		Assert.notNull(messageConverter, "MessageConverter must not be null");
 		this.amqpMessageConverter = messageConverter;
-		this.amqpTemplate.setMessageConverter(messageConverter);
+		if (!amqpTemplateExplicitlySet) {
+			((RabbitTemplate) this.amqpTemplate).setMessageConverter(messageConverter);
+		}
```

Comment:
Need to explain in javadoc that the converter will not be applied to the template if it was injected.